### PR TITLE
Address the feedback left on the Introducing Metrics APIs PR

### PR DESCRIPTION
**Pull request by tarekgh**
This change is addressing the feedback on the PR https://github.com/dotnet/runtime/pull/52685 mainly from @stephentoub and @ViktorHofer

### CODE REVIEW

**Comment by tarekgh** [comment]
@ViktorHofer I appreciate if you can have a quick look at the commit https://github.com/dotnet/runtime/pull/53324/commits/01f2ae4250f2cccdaecd82205523efb6c2955260 in this PR. This just applying your previous suggestion https://github.com/dotnet/runtime/pull/52685#pullrequestreview-658842870

**Comment by tarekgh** [comment]
CC @noahfalk

**Comment by ViktorHofer** [comment]
Yes thanks for applying my suggestion. The condition is much easier to follow now.

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace System.Diagnostics.Metrics
 #endif
     public class Meter : IDisposable
     {
-        private static List<Meter> s_allMeters = new List<Meter>();
+        private static readonly List<Meter> s_allMeters = new List<Meter>();
         private List<Instrument> _instruments = new List<Instrument>();
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs:19]: This can be readonly as well, yes?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs:19]: this one can be reassigned in the Dispose() method.

**Comment by tarekgh** [comment]
The failure in the CI leg `runtime (Build Browser wasm Release AllSubsets_Mono) ` is unrelated and tracked by the issue https://github.com/dotnet/runtime/issues/52843

**Pull request merged**